src/main/webapp/app/core/course/shared/course-base-container/course-base-container.component.ts

```diff
@@ -0,0 +1,343 @@
+import {
+    AfterViewInit,
+    ChangeDetectorRef,
+    Component,
+    EmbeddedViewRef,
+    HostListener,
+    OnDestroy,
+    OnInit,
+    QueryList,
+    TemplateRef,
+    ViewChild,
+    ViewChildren,
+    ViewContainerRef,
+    inject,
+    signal,
+} from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { HttpResponse } from '@angular/common/http';
+import { Observable, Subject, Subscription, firstValueFrom } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+import dayjs from 'dayjs/esm';
+
+import { Course, isCommunicationEnabled, isMessagingEnabled } from 'app/core/shared/entities/course.model';
+import { BarControlConfiguration } from 'app/shared/tab-bar/tab-bar';
+import { CourseStorageService } from 'app/core/course/manage/course-storage.service';
+import { CourseManagementService } from 'app/core/course/manage/course-management.service';
+import { CourseAccessStorageService } from 'app/core/course/shared/course-access-storage.service';
+import { CourseSidebarService } from 'app/core/course/overview/course-sidebar.service';
+import { MetisConversationService } from 'app/communication/metis-conversation.service';
+import { ProfileService } from 'app/core/layouts/profiles/shared/profile.service';
+import { LtiService } from 'app/shared/service/lti.service';
+import { sortCourses } from 'app/shared/util/course.util';
+import { SidebarItem } from 'app/core/course/shared/course-sidebar/course-sidebar.component';
+import { MODULE_FEATURE_ATLAS, PROFILE_IRIS, PROFILE_LOCALCI, PROFILE_LTI } from 'app/app.constants';
+
+/**
+ * Base class that contains common functionality for course container components.
+ * This abstract class is extended by CourseManagementContainerComponent and CourseOverviewComponent.
+ */
+@Component({
+    template: '',
+})
+export abstract class BaseCourseContainerComponent implements OnInit, OnDestroy, AfterViewInit {
+    protected courseManagementService = inject(CourseManagementService);
+    protected courseStorageService = inject(CourseStorageService);
+    protected route = inject(ActivatedRoute);
+    protected changeDetectorRef = inject(ChangeDetectorRef);
+    metisConversationService = inject(MetisConversationService);
+    router = inject(Router);
+    protected courseAccessStorageService = inject(CourseAccessStorageService);
+    protected profileService = inject(ProfileService);
+    protected ltiService = inject(LtiService);
+    protected courseSidebarService = inject(CourseSidebarService);
+
+    ngUnsubscribe = new Subject<void>();
+    protected closeSidebarEventSubscription: Subscription;
+    protected openSidebarEventSubscription: Subscription;
+    protected subscription?: Subscription;
+    protected profileSubscription?: Subscription;
+    protected ltiSubscription: Subscription;
+    protected loadCourseSubscription?: Subscription;
+    dashboardSubscription: Subscription;
+
+    // Common state properties using signals
```

Comment:
I do not think we need this comment
```suggestion
    // Common state properties using signals
```